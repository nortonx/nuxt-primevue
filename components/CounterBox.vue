<template>
  <div class="col-12 md:col-6 lg:col-3">
    <div class="surface-card shadow-2 p-3 border-round">
      <div class="flex justify-content-between md-3">
        <div>
          <span class="block text-500 font-medium mb-3 label">{{ props.label }}</span>
          <div class="text-900 font-medium text-xl counter">{{ props.counter }}</div>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
const props = defineProps({
  counter: Number,
  label: String
})
</script>