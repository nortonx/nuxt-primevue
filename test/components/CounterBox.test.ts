import { mount } from '@vue/test-utils'
import CounterBox from '../../components/CounterBox.vue'
import { test, expect } from 'vitest'

test('CounterBox component renders correctly', () => {
  const wrapper = mount(CounterBox)
  expect(wrapper.html()).toMatchSnapshot()
})

test('CounterBox component renders with correct props', () => {
  const wrapper = mount(CounterBox, {
    props: {
      counter: 5,
      label: "Words"
    }
  })
  expect(wrapper.find('.counter').text()).toBe('5')
  expect(wrapper.find('.label').text()).toBe('Words')
})