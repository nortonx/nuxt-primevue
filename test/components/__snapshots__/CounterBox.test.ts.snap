// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CounterBox component > should render correctly and match snapshot 1`] = `
"<div class="col-12 md:col-6 lg:col-3">
  <div class="surface-card shadow-2 p-3 border-round">
    <div class="flex justify-content-between md-3">
      <div><span class="block text-500 font-medium mb-3 label">Words</span>
        <div class="text-900 font-medium text-xl counter">0</div>
      </div>
    </div>
  </div>
</div>"
`;
